compiler/passes/convert-uast.cpp

```diff
@@ -3138,8 +3252,15 @@ struct Converter {
     // Note the module is converted so we can wire up SymExprs later
     noteConvertedSym(node, mod);
 
-    // Pop the module after converting children.
+    // Pop the module after converting children, and note the moduels used
+    // within.
     INT_ASSERT(modStack.size() > 0 && modStack.back().mod == node);
+    for (auto usedMod : modStack.back().usedModules) {
+      mod->moduleUseAdd(usedMod);
```

Comment:
The production compiler's module use list includes `use` / `import` statements from within a function. It's not clear to me how the code you have in this PR covers that (it could probably by propagating the gathered module uses, but I don't remember seeing it). Could you clarify if it handles that case?

Reply:
I'm not sure I understand what you mean. This code here is matching the production compiler behavior when detecting `(call '.' M x)` nodes:

https://github.com/chapel-lang/chapel/blob/b9d916ea104cffb83fd240a3a5b047e0f17555a0/compiler/passes/scopeResolve.cpp#L1455-L1466

I don't see any sort of recursive search, or any propagation. Since this part of the PR effectively replaces `resolveModuleCall`, I think that's all I need to do.

Reply:
Right, I think I have confused you by posting the comment on this line, when it has little to do with this line. It was a general question about the used-module tracking added in this PR to convert-uast. If we have something like

```
{
  {
    {
       use M;
     }
  }
}
```

how/why does the used module list get updated in convert-uast for this nested use case?

Reply:
Talked about this synchronously; the modules are stored on the _module_ stack, which does not get "overshadowed" by local blocks and functions. So, the innermost module gets `use` statements added to it. The _block_ stack is separate, and is used to for the (also separate) mechanic of having each block list the used modules.